tests/server/cluster/cluster_work_test.go

```diff
@@ -57,38 +56,91 @@ func (s *clusterWorkerTestSuite) TestValidRequestRegion(c *C) {
 	clusterID := leaderServer.GetClusterID()
 	bootstrapCluster(c, clusterID, grpcPDClient, "127.0.0.1:0")
 	rc := leaderServer.GetRaftCluster()
-
-	r1 := core.NewRegionInfo(&metapb.Region{
-		Id:       1,
-		StartKey: []byte(""),
-		EndKey:   []byte("a"),
-		Peers: []*metapb.Peer{{
-			Id:      1,
-			StoreId: 1,
-		}},
+	defer rc.Stop()
+
+	peers := []*metapb.Peer{{
+		Id:      2,
+		StoreId: 10,
+		Role:    metapb.PeerRole_Voter,
+	}, {
+		Id:      3,
+		StoreId: 10,
+		Role:    metapb.PeerRole_Voter,
+	}, {
+		Id:      4,
+		StoreId: 10,
+		Role:    metapb.PeerRole_Voter,
+	}}
+	r := core.NewRegionInfo(&metapb.Region{
+		Id:          1,
+		StartKey:    []byte(""),
+		EndKey:      []byte("a"),
 		RegionEpoch: &metapb.RegionEpoch{ConfVer: 2, Version: 2},
-	}, &metapb.Peer{
-		Id:      1,
-		StoreId: 1,
-	})
-	err = rc.HandleRegionHeartbeat(r1)
+		Peers:       peers,
+	}, peers[0])
+	err = rc.HandleRegionHeartbeat(r)
 	c.Assert(err, IsNil)
-	r2 := &metapb.Region{Id: 2, StartKey: []byte("a"), EndKey: []byte("b")}
-	c.Assert(rc.ValidRequestRegion(r2), NotNil)
-	r3 := &metapb.Region{Id: 1, StartKey: []byte(""), EndKey: []byte("a"), RegionEpoch: &metapb.RegionEpoch{ConfVer: 1, Version: 2}}
-	c.Assert(rc.ValidRequestRegion(r3), NotNil)
-	r4 := &metapb.Region{Id: 1, StartKey: []byte(""), EndKey: []byte("a"), RegionEpoch: &metapb.RegionEpoch{ConfVer: 2, Version: 1}}
-	c.Assert(rc.ValidRequestRegion(r4), NotNil)
-	r5 := &metapb.Region{Id: 1, StartKey: []byte(""), EndKey: []byte("a"), RegionEpoch: &metapb.RegionEpoch{ConfVer: 2, Version: 2}}
-	c.Assert(rc.ValidRequestRegion(r5), IsNil)
-	rc.Stop()
+
+	testcases := []struct {
+		region  *metapb.Region
+		checker Checker
+	}{{
+		&metapb.Region{Id: 2, StartKey: []byte("a"), EndKey: []byte("b"), Peers: peers},
+		NotNil,
+	}, {
+		&metapb.Region{Id: 1, StartKey: []byte(""), EndKey: []byte("a"), Peers: peers, RegionEpoch: &metapb.RegionEpoch{ConfVer: 1, Version: 2}},
+		NotNil,
+	}, {
+		&metapb.Region{Id: 1, StartKey: []byte(""), EndKey: []byte("a"), Peers: peers, RegionEpoch: &metapb.RegionEpoch{ConfVer: 2, Version: 1}},
+		NotNil,
+	}, {
+		&metapb.Region{Id: 1, StartKey: []byte(""), EndKey: []byte("a"), Peers: peers, RegionEpoch: &metapb.RegionEpoch{ConfVer: 1, Version: 3}},
```

Comment:
What is the difference between this line and line 91?

Reply:
`Version: 3`